src/ripple/basics/tagged_integer.h

```diff
@@ -157,6 +163,19 @@ class tagged_integer
         return tagged_integer (lhs.m_value - rhs);
     }
 
+    template <class OtherInt>
+    friend
+    typename std::enable_if <
+        std::is_integral <OtherInt>::value &&
+            sizeof (OtherInt) <= sizeof (Int),
```

Comment:
@HowardHinnant very useful. Thanks!
@nbougalis Good point.  I agree that that type of safety is beyond the scope here.  I shouldn't have used "safe" in my comment above.  Instead, I meant to convey that if you have two instances of `tagged_integer<std::int32_t, sometag>` say `a`,`b`, you should be able to all the normal binary arithmetic operations that are defined on `std::int32_t` between the instances directly without having to opt in.  So nothing about safety, but I should be able to add,subtract,multiply,divide,... pairs of `foo_int` since both are `foo_int`.  Is that what you are advising against?